### PR TITLE
Add Missing Docblock analyzer

### PR DESCRIPTION
**Pull request by ddmler**
Hey!

Type: bug fix & new feature

Link to issue: #95 

This pull request affects the following components **(please check boxes):**

* [x] Core
* [x] Analyzer
* [x] Compiler
* [ ] Control Flow Graph
* [ ] Documentation

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/ovr/phpsa/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change:
- Added missing docblock analyzer
- removed already existing missing docblock notices
- Add compile events for functions and traits

For interface compile events this is waiting for merge PR #157

### CODE REVIEW

File: src/Compiler.php [hunk 1]
```diff
@@ -95,6 +95,20 @@ public function compile(Context $context)
             $function->compile($context);
         }
 
+        foreach ($this->traits as $trait) {
+            /**
+             * @todo Configuration
+             *
+             * Ignore traits compiling from vendor
+             */
+            $checkVendor = strpos($function->getFilepath(), './vendor');
```
Comment (ovr) [inline src/Compiler.php]: `$function`, seems it's a typo, maybe `$trait`?

File: src/Analyzer/Pass/Statement/MissingDocblock.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    public function getRegister()
+    {
+        return [
+            Stmt\Class_::class,
+            Stmt\ClassMethod::class,
+            Stmt\Property::class,
+            Stmt\Function_::class,
+            Stmt\Trait_::class,
+            Stmt\Interface_::class,
+            Stmt\ClassConst::class,
```
Comment (ovr) [inline src/Analyzer/Pass/Statement/MissingDocblock.php]: About `ClassConst`, it's a hardcore time :)

Comment (ddmler) [inline src/Analyzer/Pass/Statement/MissingDocblock.php]: we can make a config setting what to activate :)

**Comment by ovr** [comment]
@ddmler 
Thanks you for this PR 👍

**Pull request merged**